### ISSUE TITLE
[A11Y] enter and space keys should activate anything that has `on:tap` or a tap handler

### ISSUE THREAD

**Issue by aghassemi** [ampproject/amphtml#8810]
Examples of what's **_broken_** currently and does not activate with enter/space keys:
- [x] amp-selector options
- [x] amp-carousel next/previous buttons
- [x] anything with `on:tap` that's not a `button` or `a` (e.g. `<div role="button" on='tap:...` tabindex=0>`)
- [x] social-share
- [x] amp-accordion (also should support up/down to expand/collapse)

**Comment by kmh287** [comment]
This definitely isn't working on desktop. Have you seen this fail on mobile?

**Comment by aghassemi** [comment]
No sure about mobile, don't have physical keyboard to test. I suspect same issue as desktop although not as important, main point of this issue is fixing it for Desktop for canonical AMP websites

**Comment by kmh287** [comment]
Did some limited testing on an emulator. Just using TAB navigation with talkback turned off and then pressing enter/space does not act as a click. 

As for mobile. talkback generates a click at the center of its focus whenever the user presses either space or enter, (I believe other screen readers behave similarly)  so it's WAI when talkback is on.

**Comment by kmh287** [comment]
@aghassemi Can you think of anything else covered by this issue that should be fixed?

**Comment by aghassemi** [comment]
@kmh287 added couple more I found to description (accordion and social-share). Most places use `<button>` which is fine.

**Comment by westonruter** [comment]
@aghassemi Has there been any discussion about adding a `preventDefault` action? In the Twenty Seventeen theme for WordPress there is a link target that has smooth scrolling behavior via using jQuery scrollTo with the following markup:

```html
<a href="#content" class="menu-scroll-down">...</a>
```

To achieve this in AMP as noted in https://github.com/ampproject/amp-wp/pull/1777 we currently have to do:

```html
<a on="tap:content.scrollTo(duration=600)" role="button" tabindex="0" class="menu-scroll-down">...</a>
```

The downside here is that the semantic `href` is lost, and it also won't work with JS turned off. Ideally we could preserve the `href` by the introduction of a `preventDefault` action:

```html
<a href="#content" on="tap:content.scrollTo(duration=600),preventDefault" class="menu-scroll-down">...</a>
```

However, is this unworkable because the event listeners are passive and thus unable to call `event.preventDefault()` in the runtime?
